Jellyfin.Api/Controllers/SessionController.cs

```diff
@@ -301,28 +301,35 @@ public class SessionController : BaseJellyfinApiController
         /// Issues a command to a client to display a message to the user.
         /// </summary>
         /// <param name="sessionId">The session id.</param>
-        /// <param name="text">The message test.</param>
-        /// <param name="header">The message header.</param>
-        /// <param name="timeoutMs">The message timeout. If omitted the user will have to confirm viewing the message.</param>
+        /// <param name="command">The <see cref="MessageCommand" /> object containing Header, Message Text, and TimeoutMs.</param>
         /// <response code="204">Message sent.</response>
         /// <returns>A <see cref="NoContentResult"/>.</returns>
         [HttpPost("Sessions/{sessionId}/Message")]
         [Authorize(Policy = Policies.DefaultAuthorization)]
         [ProducesResponseType(StatusCodes.Status204NoContent)]
         public ActionResult SendMessageCommand(
             [FromRoute, Required] string sessionId,
-            [FromQuery, Required] string text,
-            [FromQuery] string? header,
-            [FromQuery] long? timeoutMs)
+            [FromBody] MessageCommand command)
         {
-            var command = new MessageCommand
+            if (command == null)
+            {
+                throw new ArgumentException("Request body may not be null");
+            }
+
+            // Need to check if message.Text is null, since [Required] can't be applied to properties of a deserialized object.
```

Comment:
That may have been the case when json.net was the default JSON serializer, but it isn't anymore. As long as the type is nullable, [Required] will work just fine.

Reply:
I have removed the check for `command == null` and added the `[Required]` attribute to the property.

I'm currently testing to confirm that the `[Required(AllowEmptyStrings = false)]` attribute may be applied to the model property to check for null/missing/empty `Text` properties, and will update this pull request with confirmed results.

Reply:
As recommended, the [Required] attribute on the Text property of MessageCommand serves the necessary purpose.